### ISSUE TITLE
legion_defines.h is put in source directory

### ISSUE THREAD

**Issue by junghans** [StanfordLegion/legion#274]
When running cmake out of source (e.g. `cd build && cmake ..`) the source directory should stay unmodified, but currently `legion_defines.h` is created there.

I tried 4db81b8, but that seems to break something else:
```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,9 +270,9 @@ endif()
 # configure header
 #------------------------------------------------------------------------------#
 
-configure_file(${PROJECT_SOURCE_DIR}/cmake/legion_defines.h.in ${PROJECT_SOURCE_DIR}/runtime/legion_defines.h @ONLY)
-include_directories(${PROJECT_SOURCE_DIR}/runtime)
-install(FILES ${PROJECT_SOURCE_DIR}/runtime/legion_defines.h DESTINATION include)
+configure_file(${PROJECT_SOURCE_DIR}/cmake/legion_defines.h.in ${CMAKE_BINARY_DIR}/runtime/legion_defines.h @ONLY)
+include_directories(${CMAKE_BINARY_DIR}/runtime)
+install(FILES ${CMAKE_BINARY_DIR}/runtime/legion_defines.h DESTINATION include)
 
 
 #------------------------------------------------------------------------------#
```

**Comment by elliottslaughter** [comment]
I think the issue was that Legion hadn't been configured to have `${CMAKE_BINARY_DIR}/runtime` in its search path. Perhaps if you added it then it would work.

**Comment by junghans** [comment]
Isn't that what `include_directories(${CMAKE_BINARY_DIR}/runtime)` is doing?

**Comment by junghans** [comment]
@chuckatkins: Please explain that mystery.

**Comment by elliottslaughter** [comment]
@junghans I believe this is fixed now in master, can you confirm?

**Comment by junghans** [comment]
Yeah, will be fixed in #309!

**Comment by elliottslaughter** [comment]
I believe #309 was merged manually, so I'm closing this.
